Assignment 5.Rmd

```diff
@@ -40,41 +40,54 @@ We want to see if we can build a decision tree to help teachers decide which stu
 
 #Visualize our outcome variable "score"
 ```{r}
-
+hist(D1$score)
 ```
 
 #Create a categorical outcome variable based on student score to advise the teacher using an "ifelse" statement
 ```{r}
-D1$advice <- 
+# determine one-third and two-thrids score of the class
+# sort(D1$score)[nrow(D1)/3] # 0.5
+# sort(D1$score)[nrow(D1)/3*2] # 0.8333333
+D1$advice <- ifelse(D1$score <= 0.5,"intervene", ifelse(D1$score <= 0.8333333, "monitor", "no action"))
 ```
 
 #Build a decision tree that predicts "advice" based on how many problems students have answered before, the percentage of those problems they got correct and how many hints they required
 ```{r}
-score_ctree <- 
+score_ctree <- ctree(factor(advice) ~ prior_prob_count + prior_percent_correct + hints, data=D1) 
 ```
 
 #Plot tree
 ```{r}
-
+plot(score_ctree)
 ```
 
 Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
 
+Students who ask for more than 26 hints are very likely to need intervention of teachers.
+Students who ask for hints less than 26 more than 0, and have prior percent correct less than 0.595 are very likely to need intervention of teachers. 
+
+
 #Test Tree
 Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students. 
 
 ```{r}
 #Upload new data
 
-D2 <- 
+D2 <- read.csv("intelligent_tutor_new.csv")
 
 #Generate predicted advice using the predict() command for new students based on tree generated from old students
 
-D2$prediction <- 
+D2$prediction <- predict(score_ctree, D2)
 
 ``` 
 ## Part III
 Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?
+```{r}
+# As all score == 1 in the new data, we should expected there is no action needed from the teachers.
+mean(ifelse(D2$prediction == "no action", 1, 0))
+# The accuracy of the predication is 58%.
+```
+
```

Comment:
Good work! It is a good idea to check accuracy by mean for A binary variable. Maybe you could go further to comment on the precision of the model compared with baseline and think about ways to improve accuracy in your future projects.